pkg/cloudprovider/provider/aws/provider.go

```diff
@@ -640,7 +640,7 @@ func (p *provider) Create(machine *v1alpha1.Machine, userdata string) (instance.
 	return awsInstance, nil
 }
 
-func (p *provider) Delete(machine *v1alpha1.Machine, instance instance.Instance) error {
+func (p *provider) Delete(machine *v1alpha1.Machine, update cloud.MachineUpdater, instance instance.Instance) error {
```

Comment:
Could we as convention assign unused function parameters to `_` like we already did in the `fake` cloud provider?

Same for all other providers `Create` / `Delete`